dotnet/src/SemanticKernel/AI/ChatMessageGeneration/ChatHistory.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.SemanticKernel.AI.OpenAI.HttpSchema;
+
+namespace Microsoft.SemanticKernel.AI.ChatMessageGeneration;
+
+public class ChatHistory
+{
+    /// <summary>
+    /// Chat message representation
+    /// </summary>
+    public class Message
+    {
+        /// <summary>
+        /// Role of the message author, e.g. user/assistant/system
+        /// </summary>
+        public string AuthorRole { get; set; }
+
+        /// <summary>
+        /// Message content
+        /// </summary>
+        public string Content { get; set; }
+
+        /// <summary>
+        /// Create a new instance
+        /// </summary>
+        /// <param name="authorRole">Role of message author</param>
+        /// <param name="content">Message content</param>
+        public Message(string authorRole, string content)
+        {
+            this.AuthorRole = authorRole;
+            this.Content = content;
+        }
+    }
+
+    /// <summary>
+    /// List of messages in the chat
+    /// </summary>
+    public List<Message> Messages { get; set; }
```

Comment:
The setter can be marked as private to prevent setting it to null, and type can be changed to IReadonlyList if we don't need to delete history items.